bin/install-python

```diff
@@ -70,7 +76,8 @@ def main() -> int:
         ),
         Group.make(
             f'set up {py} environment',
-            (py, '-mvenv', envdir),
+            *(((py, '-mvenv', envdir) if numeric_version >= 3.4 else
+               ('pyvenv', envdir)),),
```

Comment:
this shouldn't change, `-mvenv` still works in 3.3

also no need to `*splat` as you've only got one argument (could be a simple `if ... else ...`

Reply:
```console
root@dfae863d9857:/# python3.3 -m venv testenv
root@dfae863d9857:/# testenv/bin/python --version
Python 3.3.7
```

Reply:
I mean it works in `python`, but not in the mangled version that Debian distributes.

`virtualenv` itself is a bit tricky to get working as well (probably easier for 2.7, but not for stuff like 3.1 and 3.2, which are in the support matrix for `deadsnakes` but maybe not for the GHA?).

Reply:
I'm literally using the deadsnakes python3.3 there 😕

Reply:
Ah, my mistake, the issue was actually with the `pip` command: https://github.com/dateutil/dateutil/runs/1342855084?check_suite_focus=true#step:4:140

Reply:
yeah ensurepip was added in 3.4 -- honestly I'd suggest just stop testing 3.3 and set `python_requires`

but..... if you really want to try you can do this:

```console
root@dfae863d9857:/# curl https://bootstrap.pypa.io/3.3/get-pip.py | testenv/bin/python -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1603k  100 1603k    0     0  2434k      0 --:--:-- --:--:-- --:--:-- 2433k
DEPRECATION: Python 3.3 supported has been deprecated and support for it will be dropped in the future. Please upgrade your Python.
Collecting pip<18
  Downloading https://files.pythonhosted.org/packages/0f/74/ecd13431bcc456ed390b44c8a6e917c1820365cbebcb6a8974d1cd045ab4/pip-10.0.1-py2.py3-none-any.whl (1.3MB)
    100% |################################| 1.3MB 12.4MB/s 
Collecting setuptools
  Downloading https://files.pythonhosted.org/packages/7f/e1/820d941153923aac1d49d7fc37e17b6e73bfbd2904959fffbad77900cf92/setuptools-39.2.0-py2.py3-none-any.whl (567kB)
    100% |################################| 573kB 7.6MB/s 
Collecting wheel<0.30
  Downloading https://files.pythonhosted.org/packages/8a/e9/8468cd68b582b06ef554be0b96b59f59779627131aad48f8a5bce4b13450/wheel-0.29.0-py2.py3-none-any.whl (66kB)
    100% |################################| 71kB 12.8MB/s 
Installing collected packages: pip, setuptools, wheel
Successfully installed pip-10.0.1 setuptools-39.2.0 wheel-0.29.0
root@dfae863d9857:/# test^C
root@dfae863d9857:/# ./testenv/bin/pip --version
pip 10.0.1 from /testenv/lib/python3.3/site-packages/pip (python 3.3)
```

Reply:
Do you want me to add this to this GHA, or do you want to not support Python 3.3?

I will continue testing on Python 3.3 (or abandon `dateutil`, I guess 😛)

Reply:
I mean, I'd rather not support 3.3 in GA personally

Reply:
but, if you get it working and it's not horrible then shipit